src/pages/api/public/scores.ts

```diff
@@ -37,68 +40,137 @@ export default async function handler(
-      return res.status(403).json({
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const newScore = await prisma.score.create({ data });
+
+      res.status(200).json(newScore);
+    } catch (error: unknown) {
+      console.error(error);
+      const errorMessage =
+        error instanceof Error ? error.message : "An unknown error occurred";
+      res.status(400).json({
         success: false,
-        message: "Access denied",
+        message: "Invalid request data",
+        error: errorMessage,
       });
-    // END CHECK ACCESS SCOPE
+    }
+  } else if (req.method === "GET") {
+    try {
+      if (authCheck.scope.accessLevel !== "all") {
+        return res.status(401).json({
+          success: false,
+          message:
+            "Access denied - need to use basic auth with secret key to GET scores",
+        });
+      }
 
-    const data: Prisma.ScoreCreateInput = {
-      id: obj.id ?? undefined,
-      timestamp: new Date(),
-      value: obj.value,
-      name: obj.name,
-      comment: obj.comment,
-      trace: { connect: { id: traceId } },
-      ...(obj.observationId && {
-        observation: { connect: { id: obj.observationId } },
-      }),
-    };
+      console.log(req.query);
+      const obj = ScoresGetSchema.parse(req.query); // uses query and not body
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const newScore = await prisma.score.create({ data });
+      const [scores, totalItems] = await Promise.all([
+        prisma.score.findMany({
+          where: {
+            name: obj.name ?? undefined, // optional filter
+            trace: {
+              projectId: authCheck.scope.projectId,
+              userId: obj.userId ?? undefined, // optional filter
+            },
+          },
+          include: {
+            trace: {
+              select: {
+                userId: true,
+              },
+            },
+          },
+          skip: (obj.page - 1) * obj.limit,
+          take: obj.limit,
+          orderBy: {
+            timestamp: "desc",
+          },
+        }),
+        prisma.score.count({
+          where: {
+            name: obj.name ?? undefined, // optional filter
+            trace: {
+              projectId: authCheck.scope.projectId,
+              userId: obj.userId ?? undefined, // optional filter
+            },
+          },
+        }),
+      ]);
 
-    res.status(200).json(newScore);
-  } catch (error: unknown) {
-    console.error(error);
-    const errorMessage =
-      error instanceof Error ? error.message : "An unknown error occurred";
-    res.status(400).json({
-      success: false,
-      message: "Invalid request data",
-      error: errorMessage,
-    });
+      return res.status(200).json({
+        data: scores,
+        meta: {
+          page: obj.page,
+          limit: obj.limit,
+          totalItems,
+          totalPages: Math.ceil(totalItems / obj.limit),
+        },
```

Comment:
das ist ja hier eine richtige premium API :D

Reply:
yes, needed for pagination